newIDE/app/src/BehaviorsEditor/index.js

```diff
@@ -62,6 +62,216 @@ export const useBehaviorOverridingAlertDialog = () => {
   };
 };
 
+type BehaviorConfigurationEditorInterface = {||};
+type BehaviorConfigurationEditorProps = {|
+  project: gdProject,
+  object: gdObject,
+  behavior: gdBehavior,
+  resourceManagementProps: ResourceManagementProps,
+  onBehaviorsUpdated: () => void,
+  onChangeBehaviorName: (behavior: gdBehavior, newName: string) => void,
+  onRemoveBehavior: (behaviorName: string) => void,
+  copyBehavior: (behaviorName: string) => void,
+  canPasteBehaviors: boolean,
+  pasteBehaviors: () => Promise<void>,
+  openExtension: (behaviorType: string) => void,
+|};
+
+const BehaviorConfigurationEditor = React.forwardRef<
+  BehaviorConfigurationEditorProps,
+  BehaviorConfigurationEditorInterface
+>(
+  (
+    {
+      project,
+      object,
+      behavior,
+      resourceManagementProps,
+      onBehaviorsUpdated,
+      onChangeBehaviorName,
+      onRemoveBehavior,
+      copyBehavior,
+      canPasteBehaviors,
+      pasteBehaviors,
+      openExtension,
+    },
+    ref
+  ) => {
+    const { values } = React.useContext(PreferencesContext);
+    const [expanded, setExpanded] = React.useState<boolean>(
+      !behavior.isFolded()
+    );
+    const behaviorName = behavior.getName();
+    const behaviorTypeName = behavior.getTypeName();
+
+    React.useEffect(
+      () => {
+        behavior.setFolded(!expanded);
+      },
+      [expanded, behavior]
+    );
```

Comment:
Why do you need this effect and a state? Why not using behavior.isFolded()/behavior.setFolded (and a forceUpdate when you update it)?